M2/Macaulay2/packages/VirtualResolutions.m2

```diff
@@ -137,46 +137,30 @@ resolveViaFatPoint(Ideal, Ideal, List) := ChainComplex => (J, irr, A) -> (
 -- Note: the Determinatal strategy is based on Theorem 1.3 of [Loper2019].
 --------------------------------------------------------------------
 --------------------------------------------------------------------
-isVirtual = method(Options => {Strategy => null})
-isVirtual (Ideal, ChainComplex) := Boolean => opts -> (irr, C) -> (
+isVirtual = method(TypicalValue => Boolean, Options => {Strategy => null})
+isVirtual(NormalToricVariety, ChainComplex) := opts -> (X,   C) -> isVirtual(ideal X, C)
+isVirtual(Ideal,              ChainComplex) := opts -> (irr, C) -> (
+    S := ring irr;
+    if S =!= ring C then error "isVirtual: expected objects in the same ring";
 -- if strategy "determinantal is selected, the method checks virtuality
 -- via the depth criterion on the saturated ideals of minors
+    debugInfo := if debugLevel < 1 then identity else i -> printerr("isVirtual failed at homological degree ", toString i);
     if opts.Strategy === "Determinantal" then (
         for i from 1 to length(C) do (
             if rank(source(C.dd_i)) != (rank(C.dd_i) + rank(C.dd_(i+1))) then (
-                if debugLevel >= 1 then print "isVirtual failed at homological degree " | toString i;
```

Comment:
Why does failure not result in an error?

Reply:
`isVirtual` returns a boolean. Better would be to say "virtuality failed" ...

Reply:
Hmm?  If you mean `isVirtual` returned `false`, just say that -- nothing failed.  Unless you think virtuality is a virtue and lack of virtuality is a vice.

Reply:
You're looking at the old version!

Reply:
No, I'm responding to this comment of yours: "Better would be to say "virtuality failed"".

Reply:
This conversation is moot. If you look at the current version, you'll instead see this:
https://github.com/Macaulay2/M2/blob/071bc5d6f6da278c356d3446b10ed30c1d70233c/M2/Macaulay2/packages/VirtualResolutions.m2#L147

Being a virtual resolution is equivalent to being exact as a locally-free complex of sheaves. In many areas of math people say something "fails to be exact" or "exactness fails". If the complex was found not to be virtual at that a degree, then it failed to be virtual. There's no vice is virtue in failure.